### PR TITLE
raft-engine: add the `spill-dir` config parameter

### PR DESCRIPTION
**Pull request by LykxSassinator**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Release `raft-engine.spill-dir` configurations for users who wanna uses multi disks for storing raft logs.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/18981
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -1936,6 +1936,20 @@ Raft Engine 相关的配置项。
...
++ 如果未设置此配置，则表示该目录不存在。
+
+> **注意：**
+>
+> - 该配置仅在 Raft Engine 的 `dir` 和 `spill-dir` 各自指定为**不同盘符**的情况下有效。
+> - 在配置该功能后，若想要关闭该功能，你需要在重启 TiKV **之前**执行如下操作：
+>     1. 关闭 TiKV；
+>     2. 将该目录下的所有 Raft Log 拷贝至 [`dir`](/tikv-configuration-file.md#dir) 目录下；
+>     3. 从 TiKV 配置文件中删除该配置；
+>     4. 重启 TiKV；
```
Comment (hfxsd) [inline tikv-configuration-file.md:1951]:
```suggestion
>     1. 关闭 TiKV。
>     2. 将该目录下的所有 Raft Log 复制到 [`dir`](/tikv-configuration-file.md#dir) 目录下。
>     3. 从 TiKV 配置文件中删除该配置。
>     4. 重启 TiKV。
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -1936,6 +1936,20 @@ Raft Engine 相关的配置项。
...
 + 默认值：`""`
 
+### `spill-dir` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
++ 存储 Raft 日志文件的辅助目录，当 `dir` 目录所在盘数据写满后，新的 Raft 日志将存储在该目录下。如果该目录配置后不存在，则在启动 TiKV 时创建该目录。
++ 如果未设置此配置，则表示该目录不存在。
+
+> **注意：**
+>
+> - 该配置仅在 Raft Engine 的 `dir` 和 `spill-dir` 各自指定为**不同盘符**的情况下有效。
```
Comment (hfxsd) [inline tikv-configuration-file.md:1946]:
```suggestion
> - 该配置仅在 Raft Engine 的 `dir` 和 `spill-dir` 分别指定为**不同盘符**时才有效。
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1936,6 +1936,20 @@ Raft Engine 相关的配置项。
 + 如果你的机器上有多个磁盘，建议将 Raft Engine 的数据存储在单独的磁盘上，以提高 TiKV 性能。
 + 默认值：`""`
 
+### `spill-dir` <span class="version-mark">从 v8.4.0 版本开始引入</span>
+
++ 存储 Raft 日志文件的辅助目录，当 `dir` 目录所在盘数据写满后，新的 Raft 日志将存储在该目录下。如果该目录配置后不存在，则在启动 TiKV 时创建该目录。
++ 如果未设置此配置，则表示该目录不存在。
```
Comment (hbisheng) [inline tikv-configuration-file.md:1942]: 这样更清晰一些？因为上一句刚提到“目录不存在会创建该目录”，这里再提“目录不存在”可能有点歧义。
```suggestion
+ 如果未设置此配置，则表示不启用辅助目录。
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1936,6 +1936,20 @@ Raft Engine 相关的配置项。
...
+
+> **注意：**
+>
+> - 该配置仅在 Raft Engine 的 `dir` 和 `spill-dir` 分别指定为**不同盘符**时才有效。
+> - 在配置该功能后，若想要关闭该功能，你需要在重启 TiKV **之前**执行如下操作：
```
Comment (hbisheng) [inline tikv-configuration-file.md:1947]: 如果我们知道的话，可以提一下如果不这么操作的后果？比如数据丢失，或是无法启动？
```suggestion
> - 在配置该功能后，若想要关闭该功能，你需要在重启 TiKV **之前**执行如下操作，否则会造成...：
```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1936,6 +1936,20 @@ Raft Engine 相关的配置项。
...
+>     1. 关闭 TiKV。
+>     2. 将该目录下的所有 Raft Log 复制到 [`dir`](/tikv-configuration-file.md#dir) 目录下。
```
Comment (hfxsd) [inline tikv-configuration-file.md:1949]:
```suggestion
>     2. 将 `spill-dir` 目录下的所有 Raft Log 复制到 [`dir`](/tikv-configuration-file.md#dir) 目录下。
```

**Review by hbisheng — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**